lisa/transformers/repo_package_installer.py

```diff
@@ -0,0 +1,49 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.operating_system import Posix
+from lisa.transformers.deployment_transformer import (
+    DeploymentTransformer,
+    DeploymentTransformerSchema,
+)
+
+
+@dataclass_json()
+@dataclass
+class RepoPackageInstallerTransformerSchema(DeploymentTransformerSchema):
+    install_packages: List[str] = field(default_factory=list)
+
+
+class RepoPackageInstallerTransformer(DeploymentTransformer):
+    """
+    This Transformer installs packages from a repository.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "install_repo_packages"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return RepoPackageInstallerTransformerSchema
+
+    @property
+    def _output_names(self) -> List[str]:
+        return []
+
+    def _internal_run(self) -> Dict[str, Any]:
+        runbook: RepoPackageInstallerTransformerSchema = self.runbook
+        assert isinstance(runbook, RepoPackageInstallerTransformerSchema)
+        node = self._node
+
+        if (
```

Comment:
Raise exceptions, if it's not this case.

Reply:
I raised an exception for non-Posix. But the other checks are just for `[]` or `[""]`. I didn't want to run install packages on nothing because that will upgrade all the packages. But it shouldn't be an error if the runbook has an option to install packages but provides nothing.

Reply:
It's ok not to raise an exception. It should be logged at least if nothing installed. That helps debugging issues like package string is not passed in from pipeline.